### PR TITLE
Streaming API

### LINKED ISSUE #26: Support streaming responses

### PR DESCRIPTION
**Pull request by clue**
First draft. API is subject to change in future versions.

Closes #26.

### CODE REVIEW

**Comment by asm89** [comment]
:+1: API looks a little bit clunky, but it seems like the best way given the promise API.

File: src/Io/Sender.php [hunk 1]
```diff
@@ -122,6 +122,8 @@ public function send(Request $request)
                     ));
                 }
             });
+
+            $deferred->progress(array('responseStream' => $response, 'requestStream' => $requestStream));
```
Comment (asm89) [inline src/Io/Sender.php]: nit: have constants defined on `Sender` instead of these strings? (Part of public API.)

**Comment by clue** [comment]
> :+1: API looks a little bit clunky

I'm not even going to argue on this :-) Thanks for your feedback, much appreciated!

I'd like to get this in rather sooner than later (because other libs depend on this feature).

I'm not a fan of the API, hence the statement in the README. I'm looking forward to introduce a breaking change as soon as we figure out a better API :+1:

**Comment by clue** [comment]
For the reference: This API is subject to change because it takes advantage of the dreaded promise progress API (https://github.com/reactphp/promise/issues/32). The fetch standard API (#30) might be a good alternative.

**Pull request merged**